api/k8s/v1/k8s.proto

```diff
@@ -729,6 +729,10 @@ message ConfigMap {
 
   map<string, string> annotations = 4;
   map<string, string> labels = 5;
+
+  // represents a configmaps data field which is simple
+  // key:value map
+  google.protobuf.Struct data = 6;
```

Comment:
is there a reason it's not `map<string, string>`?

also checked the underlying type https://github.com/kubernetes/api/blob/65a259a606b34fbfa8861674531aab27da4f1381/core/v1/types.go#L5895

Reply:
The thought here was that when a user needs to provide this value (say when they are creating a configmap), they will have to convert the config from json to a map[string]string. So was trying to avoid the extra conversion step for the user.
Also read that Maps cannot be repeated, optional, or required (https://developers.google.com/protocol-buffers/docs/overview#maps_features) and figured that we might want to keep this field optional

Reply:
If we are okay with making the user provide us a map for the data field I could change it to a `map<string, string>`

Reply:
ah i think i see, the user can provide `foo: {bar: baz}` and the resulting configmap is `{foo.bar: baz}`?

Reply:
Yep for example the following are some possible values for the Data field in a configmap:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: game-demo
data:
  # property-like keys; each key maps to a simple value
  player_initial_lives: "3"
  ui_properties_file_name: "user-interface.properties"

  # file-like keys
  game.properties: |
    enemy.types=aliens,monsters
    player.maximum-lives=5    
  user-interface.properties: |
    color.good=purple
    color.bad=yellow
    allow.textmode=true
```
So its not truly always just a single string mapping to another string value

Reply:
ok i think the proper type in this case is actually `map<string, protobuf.Value>` which enforces the root type is always a map.

i don't think you can have 
```yaml
data:
  - foo
  - bar
```

Reply:
Ahhh ok you might be right. Let me test that out

Reply:
Updated. I think this covers all the cases